### PR TITLE
Fix maybe add proof node

### PR DESCRIPTION
**Pull request by bendanzhentan**
See also
- https://github.com/ethereum-optimism/optimism/pull/9663
- https://github.com/bnb-chain/opbnb/pull/114

### CODE REVIEW

File: core/processor.go [hunk 1]
```diff
@@ -630,3 +634,47 @@ func indexedArgs(arguments abi.Arguments) abi.Arguments {
...
+		return size, nil
+	}
+
+	if len(proof) == 0 {
+		return proof
+	}
+
+	finalProofEl := proof[len(proof)-1]
+	size, err := SizeOfRLPList(rlp.RawValue(finalProofEl))
+	if err != nil || size != 17 {
```
Comment (welkin22) [inline core/processor.go]: Err should be printed to the log or returned to the caller. Err should not be ignored.

File: core/processor.go [hunk 2]
```diff
@@ -630,3 +634,47 @@ func indexedArgs(arguments abi.Arguments) abi.Arguments {
...
+		return proof
+	}
+
+	iter, _ := rlp.NewListIterator(rlp.RawValue(finalProofEl))
```
Comment (welkin22) [inline core/processor.go]: err should not be ignored, it should be handled properly.

File: core/processor.go (hunk 1 above)
Comment (bendanzhentan) [reply core/processor.go]: Sure. c5bdf19b9e5a3093839bf0a0c554d00d94162b7d prints out all unexpected errors. PTAL.

File: core/processor.go [hunk 3]
```diff
@@ -630,3 +634,66 @@ func indexedArgs(arguments abi.Arguments) abi.Arguments {
...
+	if len(proof) == 0 {
+		return proof
+	}
+
+	finalProofEl := proof[len(proof)-1]
+	isList, size, err := SizeOfRLPList(rlp.RawValue(finalProofEl))
+	if !isList {
+		return proof
+	}
+	if /* isList && */ err != nil {
```
Comment (welkin22) [inline core/processor.go]: I suggest not using this writing style. Comments should be separated from the actual code on different lines, or they can also be omitted. We should know that isList=true at this time.

Comment (bendanzhentan) [reply core/processor.go]: Sure. Comments were removed at 2706541e3ea7959e0390e755d1873b5efc5cf367

**Review by welkin22 — APPROVED** [review]
LGTM

**Pull request merged**